src/main/java/seedu/address/model/Entity/Issue.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.Entity;
+
+import java.util.Date;
+import java.util.Optional;
+
+public class Issue extends Entity {
+    private String description;
+    private IssueType type;
+    private boolean isCompleted;
+    private Optional<Date> time;
+
+    /**
+     * Constructor without date.
+     *
+     * @param name
+     * @param id
+     * @param description
+     * @param type
+     * @param isCompleted
+     */
+    public Issue(
+            Name name,
+            ID id,
+            String description,
+            IssueType type,
+            boolean isCompleted
+    ) {
+        super(id, name);
+        this.description = description;
+        this.type = type;
+        this.isCompleted = isCompleted;
+        this.time = Optional.empty();
+    }
+
+    /**
+     * Constructor with date.
+     *
+     * @param name
+     * @param id
+     * @param description
+     * @param type
+     * @param time
+     * @param isCompleted
+     */
+    public Issue(
+            Name name,
+            ID id,
+            String description,
+            IssueType type,
+            Optional<Date> time,
```

Comment:
Not sure if this is the right place to make a comment on this, but maybe we should standardise the kind of information we want in the Date object. This is an issue for the Storage class, since it will store the necessary information in JSON. I am suggesting "dd-MM-yyyy hh:mm:ss" for the date-time format

Reply:
Okay will `toString` it as that form then. Yes please do raise these points. Good to fix these things now. 😃